### PR TITLE
UNOMI-561 : Add merge extension logic

### PR DESCRIPTION
**Pull request by jsinovassin**
https://issues.apache.org/jira/browse/UNOMI-561

### CODE REVIEW

File: api/src/main/java/org/apache/unomi/api/schema/JsonSchemaEntity.java [hunk 1]
```diff
@@ -22,14 +22,14 @@
 /**
  * Object which represents a JSON schema stored in the persistence service
  */
-public class UnomiJSONSchema extends MetadataItem {
+public class JsonSchemaEntity extends MetadataItem {
```
Comment (dgriffon) [inline api/src/main/java/org/apache/unomi/api/schema/JsonSchemaEntity.java:25]: please rename to `JSONSchemaEntity` to match naming format

File: services/src/main/java/org/apache/unomi/services/impl/schemas/SchemaServiceImpl.java [hunk 2]
```diff
@@ -266,6 +272,69 @@ public JsonSchema getJsonSchema(String schemaId) {
...
+            JsonNode targetNode = schemaNode.at(path);
+            JsonNode nodeToAdd = entry.getValue();
+            if (targetNode.isArray()) {
+                handleArray((ArrayNode) targetNode, nodeToAdd);
+            } else if (targetNode.isObject() && nodeToAdd.isObject()) {
+                mergeIntoSchema(targetNode, nodeToAdd);
+            } else if (targetNode instanceof MissingNode) {
+                ((ObjectNode) schemaNode).set(path.replace("/", ""), nodeToAdd);
+            }
+        });
```
Comment (dgriffon) [inline services/src/main/java/org/apache/unomi/services/impl/schemas/SchemaServiceImpl.java:313]: What happen if the json node is neither an Object nor an Array?

**Review by dgriffon — CHANGES_REQUESTED** [review]

File: services/src/main/java/org/apache/unomi/services/impl/schemas/SchemaServiceImpl.java (hunk 2 above)
Comment (jsinovassin) [reply services/src/main/java/org/apache/unomi/services/impl/schemas/SchemaServiceImpl.java:313]: If it's another type, for example text, number, boolean, we don't change the value of the property in the schema. We simply skip, so the user can not override simple property.

**Review by dgriffon — APPROVED** [review]
Thank you for the contribution!

**Pull request closed without merging**